### PR TITLE
Simplified media gallery AssetDetailsProvider implementation

### LINKED ISSUE #1726: Remove AssetDetailsProviderPool
Remove `\Magento\MediaGalleryUi\Model\AssetDetailsProviderInterface` and `\Magento\MediaGalleryUi\Model\AssetDetailsProviderPool`

### PR DESCRIPTION
**Pull request by sivaschenko**
### Description (*)
Simplified media gallery AssetDetailsProvider implementation

### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/adobe-stock-integration/issues/1726

### Manual testing scenarios (*)
1. Add images from Media Gallery to Page, Block, Category and Product content
2. Go to Media Gallery (check on both embedded (opened from wysiwyg) and standalone (Content -> Media Gallery menu) Media Gallery)
3. View details of the images - the "Used In" information in View Details panel should be displayed correctly
4. Delete the images - entities, where the image is used in, should be mentioned in the deletion warning popup

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sivaschenko** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * Format image size
+     *
+     * @param int $imageSize
+     * @return string
+     */
+    private function formatSize(int $imageSize): string
+    {
+        return $imageSize === 0 ? '' : sprintf('%sKb', $imageSize / 1000);
```
Comment (lbajsarowicz) [inline app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php]: `Kb` is not a valid unit. Also - 1kB is 1024B
https://en.wikipedia.org/wiki/Kilobyte
```suggestion
        return $imageSize === 0 ? '' : sprintf('%sKB', $imageSize / 1024);
```

File: app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        foreach ($this->getUsageByEntities($id) as $type => $entities) {
+            $details[] = [
+                'name' => $this->getName($type),
+                'number' => count($entities),
+                'link' => $this->getLinkUrl($type)
+            ];
+        }
+
+        return $details;
```
Comment (lbajsarowicz) [inline app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php:68]: How about using `array_map` for that purpose?

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please review the comments.

File: app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php (hunk 1 above)
Comment (lbajsarowicz) [reply app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php]: Are you aware that when image size is `333B`, then it'll display:
image

But also sprintf can format number of digits? 
Eg. `"%.2f"`

image

Comment (lbajsarowicz) [reply app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php]: What is more - `\Zend\Validator\File\Size::toByteString` offers exactly what you have done there.

File: app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php (hunk 2 above)
Comment (sivaschenko) [reply app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php:68]: I would love to use it, but `array_map` does not provide the callback with a key

File: app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php (hunk 1 above)
Comment (sivaschenko) [reply app/code/Magento/MediaGalleryUi/Model/GetAssetDetails.php]: Thanks, went with `$size === 0 ? '' : sprintf('%.2f KB', $size / 1024);`

File: app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php (hunk 2 above)
Comment (lbajsarowicz) [reply app/code/Magento/MediaGalleryUi/Model/GetAssetUsageDetails.php:68]: OK

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
Thanks for the review @lbajsarowicz please check my updates/responses

**Review by lbajsarowicz — DISMISSED** [review]
✔️  Looks fine for me. 
We still need to wait for automated tests to finish.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7984](https://jira.corp.magento.com/browse/ENGCOM-7984) has been created to process this Pull Request

**Review by engcom-Lima — CHANGES_REQUESTED** [review]
:x: QA Failed
Steps to Reproduce:

1. Add image from Media Gallery to Page, Block, Category and Product content
2. Go to Content -> Media Gallery 
3. View details of the image
4. Click on the "Products" link in the "Used In" attribute

Expected Result: Products greed is opened
Actual Result: Page reloads (because of wrong url)
Peek 2020-08-10 15-14
Note: With Blocks, Pages and Categories behavior the same. Enable the "Add secret key to URLs" option does not affect this

**Comment by sivaschenko** [comment]
Thanks for testing, @engcom-Lima , the link URLs have been fixed

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-7984](https://jira.corp.magento.com/browse/ENGCOM-7984) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed
The "Used In" information in View Details panel is displayed correctly
Screenshot_70
Correct URLs
Peek 2020-08-10 20-08
Delete the image
Screenshot_69
Screenshot_71
All cases covered by manual testing scenario

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@magento run all tests

**Pull request merged**